src/WebRTC/SessionDescriptionHandler.js

```diff
@@ -0,0 +1,487 @@
+                self.peerConnection.addTrack(track, stream);
+              });
+            } else {
+              // Chrome 59 does not support addTrack
+              self.peerConnection.addStream(stream);
+            }
+          }, this);
+        } catch(e) {
+          self.logger.error('error adding stream');
+          self.logger.error(e);
+          return SIP.Utils.Promise.reject(e);
+        }
+        return SIP.Utils.Promise.resolve();
+      })
+      .then(function streamAdditionSucceeded() {
+        return self.createOfferOrAnswer(options.RTCOfferOptions, modifiers);
+      })
+      .then(function(sdp) {
+        return {
+          body: sdp,
+          contentType: self.CONTENT_TYPE
+        };
+      }).catch(function (e) {
+        this.session.disableRenegotiation = true;
+        throw e;
+      });
+  }},
+
+  /**
+   * Check if the Session Description Handler can handle the Content-Type described by a SIP Message
+   * @param {String} contentType The content type that is in the SIP Message
+   * @returns {boolean}
+   */
+  hasDescription: {writable: true, value: function hasDescription (contentType) {
+    return contentType === this.CONTENT_TYPE;
+  }},
+
+  /**
+   * The modifier that should be used when the session would like to place the call on hold
+   * @param {String} [sdp] The description that will be modified
+   * @returns {Promise} Promise that resolves with modified SDP
+   */
+  holdModifier: {writable: true, value: function holdModifier (sdp) {
+    if (!(/a=(sendrecv|sendonly|recvonly|inactive)/).test(sdp)) {
+      sdp = sdp.replace(/(m=[^\r]*\r\n)/g, '$1a=sendonly\r\n');
+    } else {
+      sdp = sdp.replace(/a=sendrecv\r\n/g, 'a=sendonly\r\n');
+      sdp = sdp.replace(/a=recvonly\r\n/g, 'a=inactive\r\n');
+    }
+    return SIP.Utils.Promise.resolve(sdp);
+  }},
+
+  /**
+   * Set the remote description to the underlying media implementation
+   * @param {String} sessionDescription The description provided by a SIP message to be set on the media implementation
+   * @param {Object} [options] Options object to be used by getDescription
+   * @param {MediaStreamConstraints} [options.constraints] MediaStreamConstraints https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamConstraints
+   * @param {Object} [options.peerConnectionOptions] If this is set it will recreate the peer connection with the new options
+   * @param {Array} [modifiers] Array with one time use description modifiers
+   * @returns {Promise} Promise that resolves once the description is set
+   */
+  setDescription: {writable:true, value: function setDescription (sessionDescription, options, modifiers) {
+    var self = this;
+
+    // https://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browser
+    var isFirefox = typeof InstallTrigger !== 'undefined';
+    if (!this.session.disableRenegotiation && isFirefox && this.peerConnection && this.isVideoHold(sessionDescription)) {
```

Comment:
Discussion with FF team regarding this: https://groups.google.com/forum/#!topic/mozilla.dev.media/BDnRhxZsx6I